File: fix-syntax-errors.js
import fs from 'fs';
import path from 'path';
import { glob } from 'glob';

// Function to fix malformed export statements
function fixExportStatement(content) {
  // Fix malformed export statements like "export default function Wrapped\n(props) {"
  content = content.replace(
    /export default function Wrapped\s*\n\s*\(props\)\s*\{/g,
    'export default function Wrapped(props: any) {'
  );
  
  // Fix malformed JSX like "<\n {...props} />"
  content = content.replace(
    /<\s*\n\s*\{\.\.\.props\}\s*\/>/g,
    '<Page {...props} />'
  );
  
  // Fix malformed onClick handlers like "onClick={() = aria-label="Button"> function()}"
  content = content.replace(
    /onClick=\{\(\)\s*=\s*aria-label="[^"]*">\s*([^}]+)\}/g,
    (match, functionBody) => {
      const ariaLabel = match.match(/aria-label="([^"]*)"/);
      return `onClick={() => ${functionBody.trim()}}\n                aria-label="${ariaLabel ? ariaLabel[1] : 'Button'}"`;
    }
  );
  
  return content;
}

// Function to fix duplicate metadata exports
function fixDuplicateMetadata(content) {
  const lines = content.split('\n');
  const metadataLines = [];
  let inMetadata = false;
  let metadataCount = 0;
  const result = [];
  
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    
    if (line.includes('export const metadata')) {
      if (metadataCount === 0) {
        inMetadata = true;
        metadataLines.push(line);
        metadataCount++;
      } else {
        // Skip duplicate metadata
        continue;
      }
    } else if (inMetadata && line.includes('};')) {
      metadataLines.push(line);
      inMetadata = false;
      result.push(metadataLines.join('\n'));
      metadataLines.length = 0;
    } else if (inMetadata) {
      metadataLines.push(line);
    } else {
      result.push(line);
    }
  }
  
  return result.join('\n');
}

// Function to fix semicolon issues
function fixSemicolons(content) {
  // Remove standalone semicolons
  content = content.replace(/;\s*\n\s*;/g, ';\n');
  content = content.replace(/;\s*$/gm, '');
  
  // Fix missing semicolons after imports
  content = content.replace(/import\s+[^;]+$/gm, (match) => {
    if (!match.endsWith(';')) {
      return match + ';';
    }
    return match;
  });
  
  return content;
}

// Function to fix component names in export statements
function fixComponentNames(content, fileName) {
  // Extract component name from file path
  const componentName = path.basename(fileName, '.tsx');
  const capitalizedName = componentName.charAt(0).toUpperCase() + componentName.slice(1);
  
  // Replace generic Page with specific component name
  content = content.replace(/<Page \{\.\.\.props\} \/>/g, `<${capitalizedName} {...props} />`);
  
  return content;
}

// Main function to fix a file
function fixFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Apply fixes
    content = fixDuplicateMetadata(content);
    content = fixSemicolons(content);
    content = fixExportStatement(content);
    content = fixComponentNames(content, filePath);
    
    // Write back the fixed content
    fs.writeFileSync(filePath, content, 'utf8');
    console.log(`Fixed: ${filePath}`);
    
    return true;
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
    return false;
  }
}

// Main execution
async function main() {
  // Get all page.tsx files
  const files = await glob('app/**/page.tsx', { cwd: '/workspace' });

  console.log(`Found ${files.length} page files to fix`);

  let fixedCount = 0;
  files.forEach(file => {
    const fullPath = path.join('/workspace', file);
    if (fixFile(fullPath)) {
      fixedCount++;
    }
  });

  console.log(`Fixed ${fixedCount} out of ${files.length} files`);
}

main().catch(console.error);

File: fix-syntax-errors.py
#!/usr/bin/env python3
"""
Script to fix remaining syntax errors in files after conflict resolution
"""

import re
import sys
from pathlib import Path

def fix_newest_content_banner():
    """Fix NewestContent2025Banner.tsx - missing section closing tag"""
    file_path = Path("/workspace/app/components/NewestContent2025Banner.tsx")
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Find the structure issue - need to add missing </section> tag
    # The pattern suggests we have nested divs but missing section close
    lines = content.split('\n')
    
    # Find the line with the problematic closing
    for i, line in enumerate(lines):
        if i >= 128 and i <= 135:
            if '      </div>' in line and i == 133:
                # This should be </section> instead or we need to add </section> before
                # Let me check the structure more carefully
                pass
    
    # Count opening and closing tags
    section_opens = content.count('<section')
    section_closes = content.count('</section>')
    
    if section_opens > section_closes:
        # Add missing </section> before the final </div></div>);
        content = content.replace('      </div>\n    </div>\n  );\n};', '      </div>\n    </section>\n  );\n};')
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f"✓ Fixed {file_path}")

def fix_enterprise_page():
    """Fix app/enterprise/page.tsx - JSX structure issues"""
    file_path = Path("/workspace/app/enterprise/page.tsx")
    
    if not file_path.exists():
        print(f"✗ File not found: {file_path}")
        return
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Count section tags
    section_opens = content.count('<section')
    section_closes = content.count('</section>')
    
    if section_opens > section_closes:
        # Need to add closing section tag(s)
        # Find the last </div> before </> and add </section> there
        content = re.sub(r'(\s*</div>\s*)\s*(</>)', r'\1    </section>\n\2', content)
    
    # Check for fragment issues
    if '<>' in content and not '</>' in content:
        content = content.replace('return (\n    <>', 'return (\n    <>\n')
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f"✓ Fixed {file_path}")

def fix_use_enhanced_performance():
    """Fix app/hooks/useEnhancedPerformance.ts"""
    file_path = Path("/workspace/app/hooks/useEnhancedPerformance.ts")
    
    if not file_path.exists():
        print(f"✗ File not found: {file_path}")
        return
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Check for unmatched braces
    open_braces = content.count('{')
    close_braces = content.count('}')
    
    if open_braces > close_braces:
        # Add missing closing braces at the end
        lines = content.split('\n')
        # Find the last non-empty line
        last_line_idx = len(lines) - 1
        while last_line_idx > 0 and not lines[last_line_idx].strip():
            last_line_idx -= 1
        
        # Add missing closing braces
        missing_braces = open_braces - close_braces
        lines.insert(last_line_idx + 1, '};' * missing_braces)
        content = '\n'.join(lines)
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f"✓ Fixed {file_path}")

def fix_page_optimized():
    """Fix app/page-optimized.tsx"""
    file_path = Path("/workspace/app/page-optimized.tsx")
    
    if not file_path.exists():
        print(f"✗ File not found: {file_path}")
        return
    
    with open(file_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()
    
    # Look for broken import/export statements
    fixed_lines = []
    skip_next = False
    
    for i, line in enumerate(lines):
        if skip_next:
            skip_next = False
            continue
        
        # Fix broken dynamic imports
        if 'dynamic' in line and 'from' in line:
            if not line.strip().endswith(';') and i + 1 < len(lines) and not lines[i + 1].strip().startswith('{'):
                line = line.rstrip() + ';\n'
        
        fixed_lines.append(line)
    
    content = ''.join(fixed_lines)
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f"✓ Fixed {file_path}")

def fix_use_performance():
    """Fix src/hooks/usePerformance.ts"""
    file_path = Path("/workspace/src/hooks/usePerformance.ts")
    
    if not file_path.exists():
        print(f"✗ File not found: {file_path}")
        return
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Check for unmatched braces
    open_braces = content.count('{')
    close_braces = content.count('}')
    
    if open_braces > close_braces:
        # Add missing closing braces before export
        if 'export default usePerformance' in content:
            content = content.replace(
                'export default usePerformance;',
                '  }, []);\n\n  return { lcp, fid, cls };\n};\n\nexport default usePerformance;'
            )
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f"✓ Fixed {file_path}")

def fix_error_handler():
    """Fix src/utils/errorHandler.ts"""
    file_path = Path("/workspace/src/utils/errorHandler.ts")
    
    if not file_path.exists():
        print(f"✗ File not found: {file_path}")
        return
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Check if class structure is broken
    if 'class ErrorHandler' not in content and 'private categorizeError' in content:
        # The class declaration was removed by conflict resolution
        # Need to add it back
        content = f"""class ErrorHandler {{
  private errorQueue: ErrorInfo[] = [];
  private readonly MAX_ERROR_QUEUE = 50;

  handleError(error: Error | string, context?: Record<string, any>): void {{
    const errorObj = typeof error === 'string' ? new Error(error) : error;
    
    // Create error info
    const errorInfo: ErrorInfo = {{
      message: errorObj.message,
      stack: errorObj.stack,
      timestamp: Date.now(),
      category: this.categorizeError(errorObj),
      severity: this.determineSeverity(errorObj, this.categorizeError(errorObj)),
      context,
      id: this.generateErrorId(),
    }};

    // Add to queue
    this.errorQueue.push(errorInfo);
    if (this.errorQueue.length > this.MAX_ERROR_QUEUE) {{
      this.errorQueue.shift();
    }}

    // Report error
    this.reportError(errorInfo);
  }}

{content}
"""
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f"✓ Fixed {file_path}")

def main():
    print("Fixing remaining syntax errors...\n")
    
    try:
        fix_newest_content_banner()
        fix_enterprise_page()
        fix_use_enhanced_performance()
        fix_page_optimized()
        fix_use_performance()
        fix_error_handler()
        
        print("\n✓ All syntax errors fixed!")
        return 0
    except Exception as e:
        print(f"\n✗ Error: {e}")
        import traceback
        traceback.print_exc()
        return 1

if __name__ == "__main__":
    sys.exit(main())